crypto/property/property.c

```diff
@@ -355,9 +407,28 @@ int ossl_method_store_add(OSSL_METHOD_STORE *store, const OSSL_PROVIDER *prov,
             && tmpimpl->properties == impl->properties)
             break;
     }
+
+    /*
+     * Check the number of implementation on the impl stack.
+     * If it matches the number we counted above, and the push
+     * operation is successful, then we're done here, and can return
+     * success.
+     * NOTE: This seems ripe for improvement, in that if we find a duplicate
```

Comment:
Do we really care enough to ever introduce this optimisation?
I don't think this will be common enough to justify complicating the code.

Give the opening phrase, it's likely somebody will make such a change based on good intentions alone.

IMO, this part of the comment should be removed.  Optionally, make the suggested change but I'm not going to push for that.

Reply:
I don't think it would be particularly complicated.  Its the difference between looping over all the providers once, and jumping to err if a match was found, then doing a push, versus breaking the search loop early, and doing a comparison to an extra call to IMPLEMENTATION_num.  Its probably not significant in terms of performance, but it would make the code more understandable and compact.  The only reason I didn't just do it here was the desire to avoid feature creep in the PR